Week2/todo_app/program.js

```diff
@@ -1,72 +1,203 @@
-// This is the connector (also known as driver)
-// that we can use to connect to a MySQL process
-// and access its databases.
 const mysql = require('mysql');
+const config = require('./config.json');
+var program = require('commander');
 
 class TodoModel {
-    constructor(dbConnection) {
-        this.dbConnection = dbConnection;
+    constructor(connection) {
+        this.connection = connection;
     }
 
-    // Loads all the TODOs in the database
     load(callback) {
         const selectTodoItems = "SELECT * FROM todo_items";
-        this.dbConnection.query(selectTodoItems, function(err, results, fields) {
-            if(err) {
+        this.connection.query(selectTodoItems, (err, results, fields) => {
+            if (err) {
                 callback(err);
                 return;
             }
-
             callback(null, results);
         });
     }
 
-    create(description, callback) {
-        // Write code and query to create a new TODO item
+    create(description, user_id, callback) {
+        const insertItem =
+            `INSERT INTO todo_items(text,is_completed,user_id)
+            VALUES(${connection.escape(description)} ,0 ,${connection.escape(user_id)} );
+            `;
+        this.connection.query(insertItem, (err, results, fields) => {
+            if (err) {
+                callback(err);
+                return;
+            }
+            callback(null, results);
+        });
     }
 
     update(id, description, callback) {
-        // Write code and query to update and existing TODO item
+        const updateItem =
+            `
+            UPDATE todo_items SET text = ${connection.escape(description)}
+              WHERE id = ${connection.escape(id)}
+             ;`;
+        this.connection.query(updateItem, (err, results, fields) => {
+            if (err) {
+                callback(err);
+                return;
+            }
+            callback(null, results);
+        });
     }
 
     delete(id, callback) {
-        // Write code and query to delete an existing TODO item
+        const deleteItem =
+            `
+            DELETE FROM todo_items 
+            WHERE id = ${connection.escape(id)}
+            ;`;
+        this.connection.query(deleteItem, (err, results, fields) => {
+            if (err) {
+                callback(err);
+                return;
+            }
+            callback(null, results);
+        });
     }
 
     tagTodoItem(todoItemId, tagId, callback) {
-        // Write code and query add a tag to a TODO item
+        const tagItem =
+            `
+            INSERT INTO todo_item_tag(todo_item_id,tag_id)
+            VALUES (${connection.escape(todoItemId)} , ${connection.escape(tagId)});
```

Comment:
Same comment as for everybody else - this assumes that the tag already exists in the database, like if it has been used before. How do you handle a case where the tag doesn't already exist?